hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/table/functional/TestHoodieSparkMergeOnReadTableInsertUpdateDelete.java

```diff
@@ -264,11 +263,11 @@ public void testRepeatedRollbackOfCompaction() throws Exception {
     }
   }
 
-  @Disabled("HUDI-8203")
   @ParameterizedTest
   @ValueSource(booleans = {true, false})
   public void testSimpleInsertUpdateAndDelete(boolean populateMetaFields) throws Exception {
     Properties properties = populateMetaFields ? new Properties() : getPropertiesForKeyGen();
+    properties.setProperty(HoodieTableConfig.PRECOMBINE_FIELD.key(), "timestamp");
```

Comment:
Does this mean that without preCombine field it does not work?  I've marked this test to be revisited by [HUDI-8551](https://issues.apache.org/jira/browse/HUDI-8551).

Reply:
the test data gen creates the record with an ordering field of type long. If precombine is not set, then we default to int in the filegroup reader. So then when we read the records we try to cast the long to an int which fails. timestamp field is long, so then we don't need to do any casting.
```
  public RawTripTestPayload(String jsonData, String rowKey, String partitionPath, String schemaStr) throws IOException {
    this(Option.of(jsonData), rowKey, partitionPath, schemaStr, false, 0L);
  }
```

What is also interesting is that there are 3 more constructors for RawTripTestPayload. 2 of them take in any Comparable as the ordering value. The final one sets it as an int:
```
this.orderingVal = Integer.valueOf(jsonRecordMap.getOrDefault("number", 0L).toString());
```

Reply:
We should get rid of the casting logic and do not allow preCombine field type to change.  And treat `(int) 0` as natural ordering, i.e., `(int) 0` should not be compared with real ordering value, and `(int) 0` marks commit time ordering.